ddl/ddl_api.go

```diff
@@ -492,9 +498,12 @@ func columnDefToCol(ctx sessionctx.Context, offset int, colDef *ast.ColumnDef, o
 					return nil, nil, errors.Trace(err)
 				}
 			case ast.ColumnOptionGenerated:
-				var buf = bytes.NewBuffer([]byte{})
-				v.Expr.Format(buf)
```

Comment:
We do it here, then do we need to update https://github.com/pingcap/tidb/blob/master/ddl/ddl_api.go#L2395 too?

Reply:
@zimulala Review addressed. PTAL. Thanks.

```sql
mysql> CREATE TABLE t20( first_name varchar(10), last_name varchar(10), full_name varchar(255) AS (CONCAT(first_name,' ' ,last_name) ));
Query OK, 0 rows affected (0.00 sec)

mysql> show create table t20;
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                                                                                                      |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t20   | CREATE TABLE `t20` (
  `first_name` varchar(10) DEFAULT NULL,
  `last_name` varchar(10) DEFAULT NULL,
  `full_name` varchar(255) GENERATED ALWAYS AS (concat(`first_name`, ' ', `last_name`)) VIRTUAL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

```

BEFORE:
```sql
mysql> ALTER TABLE t20 modify column full_name varchar(255) GENERATED ALWAYS AS (CONCAT(last_name,' ' ,first_name) ) VIRTUAL;
Query OK, 0 rows affected (0.01 sec)

mysql> show create table t20;
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                                                                                                      |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t20   | CREATE TABLE `t20` (
  `first_name` varchar(10) DEFAULT NULL,
  `last_name` varchar(10) DEFAULT NULL,
  `full_name` varchar(255) GENERATED ALWAYS AS (concat(`last_name`, " ", `first_name`)) VIRTUAL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

```

AFTER:
```sql
mysql> 
mysql> ALTER TABLE t20 modify column full_name varchar(255) GENERATED ALWAYS AS (CONCAT(first_name,' ' ,last_name) ) VIRTUAL;
Query OK, 0 rows affected (0.01 sec)

mysql> show create table t20;
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                                                                                                      |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t20   | CREATE TABLE `t20` (
  `first_name` varchar(10) DEFAULT NULL,
  `last_name` varchar(10) DEFAULT NULL,
  `full_name` varchar(255) GENERATED ALWAYS AS (concat(`first_name`, ' ', `last_name`)) VIRTUAL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> 
```